File: informatics/lab3/solutions/java/Task2.java
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class Task2 extends Task {
    public Task2() {
        super("task #2", """
                Дан текст. Необходимо найти в нём каждый фрагмент, где сначала идёт слово «ВТ»,
                затем не более 4 слов, и после этого идёт слово «ИТМО».
                """);
    }

    @Override
    public void run()  {
        write(operate(read()));
    }

    public String operate(String val) {
        Matcher matcher = Pattern.compile("ВТ( [а-яА-Я\\w–]+){0,4} ИТМО").matcher(val);

        StringBuilder result = new StringBuilder();
        while (matcher.find())
            result.append(matcher.group()).append("\n");

        return result.toString();
    }
}

File: informatics/lab3py/task2.py
import re

from task import Task


class Task2(Task):
    def run(self):
        self.write(self.operate(self.read()))

    def operate(self, val):
        pattern = re.compile(r'ВТ( [а-яА-Я\w–]+){0,4} ИТМО')
        matches = pattern.finditer(val)

        result = "\n".join(match.group() for match in matches)
        return result
